spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/RoutingKeyGenerator.scala

```diff
@@ -44,8 +44,12 @@ class RoutingKeyGenerator(table: TableDef, columnNames: Seq[String])
 
   private def fillRoutingKey(stmt: BoundStatement): Array[ByteBuffer] = {
     val rk = routingKey.get
-    for (i <- 0 until partitionKeyIdxs.size)
+    for (i ← partitionKeyIdxs.indices) {
+      if (stmt.isNull(partitionKeyIdxs(i)))
+        throw new NullPointerException(
```

Comment:
I don't like, but I don't want to fight for not creating it. Basically because it gives nothing but another class. The user is unable to catch it because it is thrown in executor. The driver will only get `SparkException` with the message but without the cause. So such class is useless.